src/i18n/TextCollection/i18nTextCollector.php

```diff
@@ -618,15 +618,55 @@ public function collectFromCode($content, $fileName, Module $module)
         $potentialClassName = null;
         $currentUse = null;
         $currentUseAlias = null;
+        $inVar = null;
+        $inVarText = '';
+        $stringVariables = [];
         foreach ($tokens as $token) {
             // Shuffle last token to $lastToken
             $previousToken = $thisToken;
             $thisToken = $token;
+
+            // Track string variables
+            // Store when reaching end of statement if we have content
+            if ($token === ";" && $inVar) {
+                if ($inVarText) {
+                    $stringVariables[$inVar] = $inVarText;
+                }
+                $inVar = null;
+                continue;
+            }
+            // End track string variables
+
+            // Not all tokens are returned as an array.
+            // If a token is not variable, but instead it is one particular constant string, it is returned as a string instead.
+            // You don't get a line number.
+            // This is the case for braces( "{", "}"), parentheses ("(", ")"), brackets ("[", "]"), comma (","), semi-colon (";")...
             if (is_array($token)) {
                 list($id, $text, $lineNo) = $token;
                 // minus 2 is used so the the line we get corresponds with what number token_get_all() returned
                 $line = $lines[$lineNo - 2] ?? '';
 
+                // Ignore whitespace
+                if ($id === T_WHITESPACE) {
+                    continue;
+                }
+
+                // Track string variables
+                if (!$inTransFn) {
+                    if ($id === T_VARIABLE) {
+                        $inVar = $text;
+                        $inVarText = '';
+                        continue;
+                    }
+                    if ($id === T_CONSTANT_ENCAPSED_STRING && $inVar && $text) {
```

Comment:
```suggestion
                    if ($id === T_CONSTANT_ENCAPSED_STRING && $inVar !== null && $text !== null) {
```
As per https://github.com/silverstripe/silverstripe-framework/pull/11521#discussion_r1903933994

Reply:
perfect